module/VuFind/src/VuFind/Controller/TurnstileController.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+/**
+ * Turnstile Controller
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2024.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+namespace VuFind\Controller;
+
+use Laminas\Log\LoggerAwareInterface;
+use Laminas\ServiceManager\ServiceLocatorInterface;
+use VuFind\Crypt\HMAC;
+use VuFind\Log\LoggerAwareTrait;
+use VuFind\RateLimiter\RateLimiterManager;
+use VuFind\RateLimiter\Turnstile\Turnstile;
+
+/**
+ * Controller Cloudflare Turnstile access checks.
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+class TurnstileController extends AbstractBase implements
+    LoggerAwareInterface
+{
+    use LoggerAwareTrait;
+
+    protected $hashKeys = ['siteKey', 'policyId', 'destination'];
```

Comment:
We should probably put a `@var` comment on this.

Reply:
Yes, I don't get why the automated checks that find missing method param docs don't find them on class params!

Reply:
Yeah, this baffles me too. Maybe there's some PHP_CodeSniffer setting we could add to address this, but I haven't been able to find it in the past. I believe that if you look at the generated PHPDoc, it does include a list of missing property annotations, so that's one way to catch things (but not a particularly helpful one).

Reply:
Just looked at PHP_CodeSniffer and I can't find it either.